src/Session.java

```diff
@@ -1,72 +1,128 @@
-                String guess = input.nextLine();
+      System.out.println(currentPlayer + ": please enter a guess");
 
-                switch (word.guess(guess)) {
-                    case CORRECT_GUESS:
-                        System.out.println("Correct guess!");
-                        break guessLoop;
-                    case INCORRECT_GUESS:
-                        lives--;
-                        System.out.println("Incorrect guess, lose a live");
-                        break guessLoop;
-                    case UNGUESSABLE_CHAR:
-                        System.out.println("Character cannot be guessed, please pick another");
-                        break;
-                    case COMPLETED_WORD:
-                        players.forEach(p -> p.incrementScore(5));
-                        currentPlayer.incrementScore(5);
-                        leader.decrementScore(3);
-                        System.out.println(word);
-                        System.out.println("Game won! 10 points to " + currentPlayer + ", 5 to others, leader loses 3");
-                        return;
-                    case CHAR_ALREADY_GUESSED:
-                        System.out.println("Character already guessed, please pick another");
-                        break;
-                }
-            }
+      guessLoop:
+      while (true) {
+        String guess = input.nextLine();
 
+        switch (word.guess(guess)) {
+          case CORRECT_GUESS:
+            System.out.println("Correct guess!");
+            break guessLoop;
+          case INCORRECT_GUESS:
+            lives--;
+            System.out.println("Incorrect guess, lose a live");
+            leaderBoard = updateLeaderBoard();
+            break guessLoop;
+          case UNGUESSABLE_CHAR:
+            System.out.println("Character cannot be guessed, please pick another");
+            break;
+          case COMPLETED_WORD:
+            players.forEach(p -> p.incrementScore(5));
+            currentPlayer.incrementScore(5);
+            leader.decrementScore(3);
             System.out.println(word);
-            currentPlayerIndex = (currentPlayerIndex + 1) % players.size();
+            System.out.println(
+                "Game won! 10 points to " + currentPlayer + ", 5 to others, leader loses 3");
+            return;
+          case CHAR_ALREADY_GUESSED:
+            System.out.println("Character already guessed, please pick another");
+            break;
         }
+      }
 
-        System.out.println("No lives left, round lost");
-        leader.incrementScore(10);
+      System.out.println(word);
+      System.out.println("LeaderBoard:");
+      System.out.println(leaderBoard);
+
+      currentPlayerIndex = (currentPlayerIndex + 1) % players.size();
     }
 
+    System.out.println("No lives left, round lost");
+    System.out.println(word);
+    leader.incrementScore(10);
+  }
 
+  private String updateLeaderBoard() {
+    StringBuilder leaderBoard = new StringBuilder();
```

Comment:
Why do you need this StringBuilder at all? You are literally accessing a String at a single array location and returning it. 

Also, I don't understand the purpose of the index variable. I'm assuming it keeps track of which hangman graphic we should be currently showing. But that's based on the number of lives the team has left, no? Then we don't need to store any extra index variables, and this whole function can be a one-liner:

return indexLeaderBoard[maxLives - lives];

Of course this means you need to initialise maxLives in the constructor. An easier solution would be to reverse the order of the indexLeaderBoard array.